protocol-test-util/src/test/java/io/camunda/zeebe/test/broker/protocol/record/ProtocolFactoryTest.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.test.broker.protocol.record;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.RecordAssert;
+import io.camunda.zeebe.protocol.record.ValueType;
+import io.camunda.zeebe.protocol.util.ValueTypeMapping;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+@Execution(ExecutionMode.CONCURRENT)
+final class ProtocolFactoryTest {
+  @Test
+  void shouldUseADifferentSeedOnConstruction() {
+    // given
+    final var factoryA = new ProtocolFactory();
+    final var factoryB = new ProtocolFactory();
+
+    // when
+    final var seedA = factoryA.getSeed();
+    final var seedB = factoryB.getSeed();
+
+    // then
+    assertThat(seedA).isNotEqualTo(seedB);
+  }
+
+  @Test
+  void shouldGenerateRecordDeterministically() {
+    // given
+    final var factoryA = new ProtocolFactory();
+    final var factoryB = new ProtocolFactory(factoryA.getSeed());
+
+    // when
+    final var recordA = factoryA.generateRecord();
+    final var recordB = factoryB.generateRecord();
+
+    // then
+    assertThat(recordA).isEqualTo(recordB);
+  }
+
+  @Test
+  void shouldGenerateRecordsDeterministically() {
```

Comment:
🔧 If possible, please add a test case that records are generated randomly, i.e. different seeds lead to different records. Not sure whether such a test can be made deterministically. I imagine that some different seeds might lead to the same records.